src/platform/tls_openssl.c

```diff
@@ -0,0 +1,3457 @@
+/*++
+
+    Copyright (c) Microsoft Corporation.
+    Licensed under the MIT License.
+
+Abstract:
+
+    Implements the TLS functions by calling OpenSSL.
+
+--*/
+#include "platform_internal.h"
+
+#include "openssl/opensslv.h"
+#if OPENSSL_VERSION_MAJOR >= 3
+#define IS_OPENSSL_3
```

Comment:
We no longer support any version less than 3, so this can be removed with all the older stuff now.